src/Generator/Program.fs

```diff
@@ -259,12 +271,44 @@ let main argv =
     let typecheckedFunctions = looslyTypedFunctionsToTypedFunctions enumMap functions
     printfn "overall correct function specifications: %d" typecheckedFunctions.Length
     let openGlSpecVersions = getEnumCasesAndCommandsPerVersion test
-    for currOpenGL_Spec in openGlSpecVersions do
-        generateEnums enums
-        generateInterface typecheckedFunctions
-        generateStaticClass typecheckedFunctions
-        generateDummyTypes()
+    let basePath = "./"
+    let inline writeToFile openGlVersion topic content =
+        let pathToFile = basePath </> openGlVersion
+        let fileName = topic + ".cs"
+        Directory.CreateDirectory pathToFile |> ignore
+        File.WriteAllText(pathToFile </> fileName, content)
+    openGlSpecVersions 
+    |> Array.Parallel.iter (fun currOpenGL_Spec ->
+        let inline writeToFile topic content = writeToFile currOpenGL_Spec.version topic content
+        let enums =
+            enums
+            |> Array.Parallel.choose(fun enum ->
+                let cases = enum.cases |> Array.filter (fun case -> currOpenGL_Spec.enumCases.Contains case.name)
+                if cases.Length > 0 then
+                    { enum with
+                        EnumGroup.cases = cases }
+                    |> Some
+                else None
+            )
+        let typecheckedFunctions =
+            typecheckedFunctions
+            |> Array.filter(fun func -> currOpenGL_Spec.functions.Contains func.name)
+        Formatting.generateEnums enums currOpenGL_Spec
+        |> writeToFile "Enums"
+        Formatting.generateInterface typecheckedFunctions currOpenGL_Spec
+        |> writeToFile "Interface"
+        Formatting.generateStaticClass typecheckedFunctions currOpenGL_Spec
+        |> writeToFile "StaticClass"
+        Formatting.generateDummyTypes currOpenGL_Spec
+        |> writeToFile "DummyTypes"
+        printfn "Done writing OpenGL Version %s files." currOpenGL_Spec.version
+    )
     //for func in typecheckedFunctions |> Array.take 100 do
     //    printfn "%A" func
 
+    printfn "Generating files took %s seconds" (startTime.Elapsed.Seconds |> string)  
+
+    printfn "Please press any key..."
+    System.Console.ReadKey() |> ignore
```

Comment:
Are these 2 lines necessary? it terminates here anyway

Reply:
locally removed